docs/sources/developers/plugins/backend/debug-your-backend-plugin.md

```diff
@@ -0,0 +1,69 @@
++++
+title = "Debug your backend plugin"
+keywords = ["grafana", "plugins", "backend", "plugin", "backend-plugins", "documentation"]
+type = "docs"
++++
+
+# Debug your backend plugin
+
+This guide explains how to attach a debugger to a running backend plugin.
+
+A backend plugin runs in its own process, separate from grafana-server. This makes it difficult to debug the plugin the same way you would debug the grafana-server process.
+
+You can read more about the how plugin processes are managed in the [go-plugin](https://github.com/hashicorp/go-plugin#architecture) project page.
+
+Fortunately, you can debug your plugin by attaching the [Delve](https://github.com/go-delve/delve) debugger for Go to the subprocess running your plugin.
+
+You can add more information available while debugging by building your plugin with additional flags.
+
+1. Disable compiler optimizations and inlining by adding `-gcflags=all="-N -l"` when you build your plugin.
+
+   ```
+   go build -gcflags=all="-N -l" -o ./dist/my-datasource_linux_amd64 ./backend
+   ```
+
+1. Restart grafana-server or kill running plugin instance to restart plugin.
+
+   ```
+   pkill my-datasource
+   ```
+
+1. Attach to plugin process.
+
+   ```
+   dlv attach <pid>
+   ```
+
+## Run Delve in headless mode
```

Comment:
I suggest adding a brief intro, perhaps explaining why we run Delve in headless mode rather than normal mode.

Reply:
@alexanderzobnin Do you mind elaborating on why the user would want to run Delve in headless mode?